__test__/api/comments/updateSingleCommentFlags.test.js

```diff
@@ -0,0 +1,110 @@
+const app = require("../../../server");
+const CommentModel = require("../../../models/comments");
+const commentHandler = require("../../../utils/commentHandler");
+const supertest = require("supertest");
+const request = supertest(app);
+
+describe("PATCH /comments/:commentId/flag", () => {
+  let sampleComment, sampleComment2;
+
+  beforeEach(async () => {
+    const comment = new CommentModel({
+      refId: "94736880",
+      applicationId: global.application._id,
+      ownerId: "myuseremail@email.com",
+      content: "this is a very bad comment",
+      origin: "myuseremail@email.com",
+    });
+
+    const comment2 = new CommentModel({
+      refId: "576897564",
+      applicationId: global.application._id,
+      ownerId: "anotheruseremail@email.com",
+      content: "this is another comment ",
+      origin: "myuseremail@email.com",
+    });
+
+    const commentOne = await comment.save();
+    const commentTwo = await comment2.save();
+
+    sampleComment = commentHandler(commentOne);
+    sampleComment2 = commentHandler(commentTwo);
+  });
+
+  afterEach(async () => {
+    await CommentModel.findByIdAndDelete(sampleComment.commentId);
+    await CommentModel.findByIdAndDelete(sampleComment2.commentId);
+
+    sampleComment = null;
+    sampleComment2 = null;
+  });
+
+  test("Flag a single comment. Should return 200", async () => {
+    const res = await request
```

Comment:
I see that you already check the state of the DB after the request to see if the request did actually make the update it was required to. That's great.

But you should also check the state of the DB before the request to see if the comment is exactly what we expect it to be before the update. This way we can be certain that only our update made a change. Because if you don't do the initial state change how can we be sure that the update was ours or a previously made one. I hope you understand what I mean.